### PR TITLE
NIS-13912: added project structure

### PR DESCRIPTION
**Pull request by westcrime**
Added project structure.

### CODE REVIEW

File: Offices/Offices.API/Offices.API.csproj [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    <ImplicitUsings>enable</ImplicitUsings>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Swashbuckle.AspNetCore" Version="6.6.2" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <ProjectReference Include="..\Offices.Application\Offices.Application.csproj" />
+    <ProjectReference Include="..\Offices.Data\Offices.Data.csproj" />
```
Comment (Glipchik) [inline Offices/Offices.API/Offices.API.csproj:15]: Why do u have reference to Data project in API?

File: Offices/Offices.API/Program.cs [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+// Add services to the container.
+
+builder.Services.AddControllers();
+// Learn more about configuring Swagger/OpenAPI at https://aka.ms/aspnetcore/swashbuckle
+builder.Services.AddEndpointsApiExplorer();
+builder.Services.AddSwaggerGen();
+
+// Adding services to container
+builder.Services.AddScoped<IOfficeService, OfficeService>();
```
Comment (Glipchik) [inline Offices/Offices.API/Program.cs:14]: Move this to the separate extension method for ServicesProvider in application layer where you will have all dependencies in the same place for the specific project. And call this extension in program.cs

File: Offices/Offices.API/Program.cs [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+builder.Services.AddScoped<IOfficeRepository, OfficeRepository>();
```
Comment (Glipchik) [inline Offices/Offices.API/Program.cs:15]: And here it is the same

**Review by Glipchik — CHANGES_REQUESTED** [review]

File: Offices/Offices.Application/Extensions/RepositoriesExtension.cs [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Offices.Data.Repositories;
+
+namespace Offices.Application.Extensions
+{
+    public static class RepositoriesExtension
```
Comment (Glipchik) [inline Offices/Offices.Application/Extensions/RepositoriesExtension.cs:12]: This should be in the Data layer like everything related to the interaction with the db

File: Offices/Offices.API/Offices.API.csproj (hunk 1 above)
Comment (westcrime) [reply Offices/Offices.API/Offices.API.csproj]: Fixed

File: Offices/Offices.API/Program.cs (hunk 2 above)
Comment (westcrime) [reply Offices/Offices.API/Program.cs]: Fixed

File: Offices/Offices.API/Program.cs [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+
+// Add services to the container.
+
+builder.Services.AddControllers();
+// Learn more about configuring Swagger/OpenAPI at https://aka.ms/aspnetcore/swashbuckle
+builder.Services.AddEndpointsApiExplorer();
+builder.Services.AddSwaggerGen();
+
+// Configuring container
+RepositoriesExtension.AddRepositories(builder.Services);
```
Comment (Glipchik) [inline Offices/Offices.API/Program.cs:14]: Extension methods should be called like this: 
`builder.Services.AddRepositories`
And not using the class that contains it

File: Offices/Offices.API/Program.cs (hunk 3 above)
Comment (westcrime) [reply Offices/Offices.API/Program.cs]: Fixed

**Review by Glipchik — CHANGES_REQUESTED** [review]

File: Offices/Offices.Application/Extensions/RepositoriesExtension.cs (hunk 4 above)
Comment (westcrime) [reply Offices/Offices.Application/Extensions/RepositoriesExtension.cs]: Fixed

File: Offices/Offices.API/Program.cs (hunk 5 above)
Comment (westcrime) [reply Offices/Offices.API/Program.cs]: Fixed

**Pull request merged**